Added loggers for App Engine flexible environment.

tmatsuo opened the pull request:
Fixes #9 

This handler will format the logging message into a json format for a better presentation on the Cloud logging UI, in particular, with this format, you can filter the logs by severity, and also you should be able to bundle logs in a single request (unfortunately it doesn't work for me yet).

Downside is this PR adds monolog/monolog dependency. If you don't like big dependencies, please let me know. Potentially we can put these loggers in a different repo.

dwsupplee commented on src/Logger/AppEngineFlexHandler.php:
```diff
@@ -0,0 +1,63 @@
+     * @param Boolean $bubble Whether the messages that are handled can bubble up the stack or not.
+     * @param int|null $filePermission Optional file permissions (default (0640) are only for owner read/write).
+     * @param Boolean $useLocking Try to lock log file before doing any writes.
+     * @param resource|string $stream
```
Would you be able to also document the null type here? Only mentioning to keep consistent with line 30. :)

dwsupplee commented on tests/Logger/AppEngineFlexHandlerTest.php:
```diff
@@ -0,0 +1,55 @@
+class AppEngineFlexHandlerTest extends \PHPUnit_Framework_TestCase
+{
+    private $path;
+    private $handler;
```
It looks like this may need to be $log

dwsupplee commented:
This looks good to me! Eventually we will be adding support for [the logging API](https://cloud.google.com/logging/docs/) - I'm thinking once we get there we would use the namespace 'Logging' to keep consistent with the name of the service. 

What are your thoughts on that, do you think a 'Logger' and 'Logging' namespace would make sense, or could we house them under one?

tmatsuo commented:
@dwsupplee First I have addressed your comments in the second commit. Yes, logging API library code should be under `Logging` and I would rather separate them. `Logging` is the client lib for accessing Logging API, and `Logger` is for utilities for better user-land logging.

WDYT?

dwsupplee commented:
That makes sense, sounds like it can work well to me

tmatsuo commented:
I just squashed the commits.

cedricziel commented:
Can we have a tag for this?

dwsupplee commented:
@cedricziel [all set](https://github.com/GoogleCloudPlatform/gcloud-php/releases/tag/v0.2.0)

cedricziel commented on composer.json:
```diff
@@ -26,6 +26,7 @@
     "google/auth": "0.7",
     "guzzlehttp/guzzle": "~5.2|~6.0",
     "guzzlehttp/psr7": "1.2.*",
+    "monolog/monolog": "~1",
```
I'm not super concerned re: hard dependency here. Maybe it's more elegant to use a "suggest" section since it's in fact no hard dependency to the library, just for a specific feature (as of now).

cedricziel commented on src/Logger/AppEngineFlexHandler.php:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * Get the default formatter.
+     *
+     * @return FormatterInterface
```
Import for `FormatterInterface` is missing.

dwsupplee replied:
Good catch. Pushing up a fix now.

JustinBeckwith commented on src/Logger/AppEngineFlexFormatter.php:
```diff
@@ -0,0 +1,64 @@
+        list($usec, $sec) = explode(" ", microtime());
+        $usec = (int)(((float)$usec)*1000000000);
+        $sec = (int)$sec;
+        $payload = [
```
@tmatsuo I know this was already merged - but is there a way we can pass a trace Id to provide request correlation?

tmatsuo replied:
Trace id is automatically added based on the $_SERVER['HTTP_X_CLOUD_TRACE_CONTEXT'] on App Engine flex. Are you talking about other platforms?

JustinBeckwith replied:
*hangs head* and next time I'll read more closely :)
